pkg/transportrequest/gitutils_test.go

```diff
@@ -0,0 +1,175 @@
+
+		tests := [][]string{
+			[]string{
+				"straight forward",
+				"this is a commit with TransportRequestId\n\nThis is the first line of the message body\nTransportRequest: 12345678",
+				"12345678",
+			},
+			[]string{
+				"trailing spaces after our value",
+				"this is a commit with TransportRequestId\n\nThis is the first line of the message body\nTransportRequest: 12345678  ",
+				"12345678",
+			},
+			[]string{
+				"trailing text after our value",
+				"this is a commit with TransportRequestId\n\nThis is the first line of the message body\nTransportRequest: 12345678 aaa",
+			},
+			[]string{
+				"leading whitespace before our label",
+				"this is a commit with TransportRequestId\n\nThis is the first line of the message body\n   TransportRequest: 12345678",
+				"12345678",
+			},
+			[]string{
+				"leading text before our label",
+				"this is a commit with TransportRequestId\n\nThis is the first line of the message body\naaa TransportRequest: 12345678",
+			},
+			[]string{
+				"whitespaces before column",
+				"this is a commit with TransportRequestId\n\nThis is the first line of the message body\nTransportRequest  : 12345678",
+				"12345678",
+			},
+			[]string{
+				"no whitespaces after column",
+				"this is a commit with TransportRequestId\n\nThis is the first line of the message body\nTransportRequest  :12345678",
+				"12345678",
+			},
+			[]string{
+				"two times the same id in the same commit",
+				"this is a commit with TransportRequestId\n\nThis is the first line of the message body\nTransportRequest : 12345678\nTransportRequest : 12345678",
+				"12345678",
+			},
+			[]string{
+				// we report the ids, this is basically an invalid state, but needs to be filtered out by the caller
+				"two different ids in the same commit",
+				"this is a commit with TransportRequestId\n\nThis is the first line of the message body\nTransportRequest : 12345678\nTransportRequest : 87654321",
+				"12345678", "87654321",
+			},
+		}
+
+		for _, testConfig := range tests {
+			runTest(testConfig)
+		}
+	})
+
+	t.Run("multi commit tests", func(t *testing.T) {
+
+		t.Run("two different ids in different commits", func(t *testing.T) {
+			commitIter := &commitIteratorMock{
+				commits: []object.Commit{
+					object.Commit{
+						Hash:    plumbing.NewHash("3434343434343434343434343434343434343434"),
+						Message: "this is a commit with TransportRequestId\n\nThis is the first line of the message body\nTransportRequest: 12345678",
+					},
+					object.Commit{
+						Hash:    plumbing.NewHash("1212121212121212121212121212121212121212"),
+						Message: "this is a commit with TransportRequestId\n\nThis is the first line of the message body\nTransportRequest: 87654321",
+					},
+				},
+			}
+			labels, err := FindLabelsInCommits(commitIter, "TransportRequest")
+			if assert.NoError(t, err) {
+				expected := []string{"12345678", "87654321"}
+				if assert.Len(t, labels, len(expected)) {
+					assert.Subset(t, expected, labels)
```

Comment:
If the order of the ids is of importance, it should be checked.

Reply:
IMO the order is not important. It is technically a Set. Here in go we have a string slice which is technically a List.

Just for clarification: Set: no duplicates, order not important, List, duplicates, order is important.

Would be possible to use a map in go, where the values are not important. But does not look very idiomatic to me ...

Reply:
@rodibrin : the order of the returnd labels (e..g transportRequestIds, changeDocumentIds) is now always ascending.